scripts/azureml-assets/azureml/assets/config.py

```diff
@@ -141,6 +154,139 @@ def release_paths(self) -> List[Path]:
         return release_paths
 
 
+class ModelConfig(Config):
+    """
+
+    Example:
+    
+        name: bert-base-uncased
+        path: # should be string or should contain package details
+            package: # if package not specificed , path would be be used like os path object.
+                url: https://huggingface.co/bert-base-uncased
+                commit_hash: 5546055f03398095e385d7dc625e636cc8910bf2
+        publish:
+            type: mlflow_model # could be one of (custom_model, mlflow_model, triton_model)
+            flavor: hftransformers # flavors should be specificed only for mlflow_model and model would be published in this flavor
+            tags: # Key value pairs which will be published as tags in the registry 
+                isv : huggingface
+                task: fill-mask
+    """
+
+    @property
+    def name(self) ->str: 
+        return self._yaml.get("name")
+
+    @property
+    def path(self) -> object:
+        return self._yaml.get("path")
+
+    @property
+    def package(self) -> Dict[str,str]:
+        return self.path.get("package") if type(self.path()!="str") else None
+        
+    @property
+    def commit_hash(self) -> str:
+        return self.package.get("commit_hash") if self.package() !=None else None
+
+    @property
+    def url(self) -> str:
+        return self.package.get("url") if self.package() !=None else None
+
+    @property
+    def publish(self) -> Dict[str,object]:
+        return self._yaml.get("publish")
+
+    @property
+    def type(self) -> str:
+        return self.publish.get("type")
+    
+    @property
+    def flavor(self) -> str:
+        return self.publish.get("flavor") if self.publish.get("flavor") else None
+
+    @property
+    def tags(self) -> Dict[str,str]:
+        return self.publish("tags") if self.publish.get("tags") else {}
+    
+    @property
+    def task_name(self) -> str:
+        return self.tags.get("task") if self.tags.get("task") else None
+
+    @property
+    def model_dir(self) -> str:

```

Comment:
Please use `Path` objects instead of strings for file/directories, and move this function back to `asset_publish.py`.

Reply:
Moved to asse_publish.